### PR TITLE
Don't initialize RenderFrameData when connection to renderer is closed

### PR DESCRIPTION
**Pull request by simonhong**
Close https://github.com/brave/brave-browser/issues/277

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions

### CODE REVIEW

**Comment by simonhong** [comment]
However, I'm not sure this is right solution.

**Comment by bbondy** [comment]
I don't know either, @bridiver ?

**Comment by bbondy** [comment]
let's go with it and see if there's a problem

File: components/brave_shields/browser/brave_shields_web_contents_observer.cc [hunk 1]
```diff
@@ -100,6 +100,9 @@ BraveShieldsWebContentsObserver::BraveShieldsWebContentsObserver(
 
 void BraveShieldsWebContentsObserver::RenderFrameCreated(
     RenderFrameHost* rfh) {
+  if (!rfh->IsRenderFrameLive())
```
Comment (bridiver) [inline components/brave_shields/browser/brave_shields_web_contents_observer.cc]: we just went through this in muon - see https://github.com/brave/muon/commit/56d63560939ef407eaddea685c235b5b166b1475
and all those calls to `extensions::ExtensionApiFrameIdMap` should be removed

**Comment by bridiver** [comment]
I think this should be reverted - see my comments

**Comment by bbondy** [comment]
Reverted here: https://github.com/brave/brave-core/pull/143

**Pull request merged**